### ISSUE TITLE
need debug/error log when parsing fails

### ISSUE THREAD

**Issue by sdole** [awslabs/amazon-kinesis-agent#93]
I am debugging my custom regex pattern for LOGTOJSON with a custom matchPattern and customFieldNames. There does not seem to be any logging about what could be going wrong. Can you add some debug logging [in the Parser](https://github.com/awslabs/amazon-kinesis-agent/blob/master/src/com/amazon/kinesis/streaming/agent/processing/processors/LogToJSONDataConverter.java#L74)?

**Comment by chaochenq** [comment]
Yes. Thanks for reporting this. I will add more logging for debugging.

**Comment by sdole** [comment]
In this case, it would be nice to have a WARN or ERROR log when conversion fails due to a regex issue and DEBUG with exact data or failure details. Thank you.

**Comment by chaochenq** [comment]
The regex exception might not be an actual error though. You can specify matchPattern to make agent filter out the records that don't match the pattern. 

I think log at DEBUG level makes more sense, otherwise WARN and ERROR could be quite spammy since this happens per record base.

**Comment by sdole** [comment]
That is right. I had not thought about that.
